### PR TITLE
Main chain start round notarization in sovereign

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Main chain start round notarization in sovereign
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: cmd/sovereignnode/sovereignNodeRunner.go [hunk 1]
```diff
@@ -1670,6 +1671,7 @@ func createWhiteListerVerifiedTxs(generalConfig *config.Config) (process.WhiteLi
 func createSovereignWsReceiver(
 	dataPool dataRetriever.PoolsHolder,
 	config *sovereignConfig.NotifierConfig,
+	startRound uint64,
```
Comment (SebastianMarian) [inline cmd/sovereignnode/sovereignNodeRunner.go:1674]: mainChainStartRound?

File: config/sovereignConfig.go [hunk 2]
```diff
@@ -4,4 +4,9 @@ package config
 type SovereignConfig struct {
 	ExtendedShardHdrNonceHashStorage StorageConfig
 	ExtendedShardHeaderStorage       StorageConfig
+	MainChainNotarization            MainChainNotarization `toml:"MainChainNotarization"`
+}
+
+type MainChainNotarization struct {
```
Comment (SebastianMarian) [inline config/sovereignConfig.go:10]: Missing comment

File: config/sovereignConfig.go [hunk 3]
```diff
@@ -4,4 +4,9 @@ package config
 type SovereignConfig struct {
 	ExtendedShardHdrNonceHashStorage StorageConfig
 	ExtendedShardHeaderStorage       StorageConfig
+	MainChainNotarization            MainChainNotarization `toml:"MainChainNotarization"`
+}
+
```
Comment (raduchis) [inline config/sovereignConfig.go:9]: missing comment

File: cmd/sovereignnode/config/sovereignConfig.toml [hunk 4]
```diff
@@ -24,3 +24,7 @@
         BatchDelaySeconds = 2
         MaxBatchSize = 100
         MaxOpenFiles = 10
+
+[MainChainNotarization]
+    # This defines the starting round from which all sovereign chain nodes should starting notarizing main chain headers
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply cmd/sovereignnode/config/sovereignConfig.toml:29]: Right, renamed it to MainChainNotarizationStartRound

File: cmd/sovereignnode/incomingHeader/incomingHeaderProcessor.go [hunk 5]
```diff
@@ -20,11 +21,13 @@ type ArgsIncomingHeaderProcessor struct {
 	TxPool      TransactionPool
 	Marshaller  marshal.Marshalizer
 	Hasher      hashing.Hasher
+	StartRound  uint64
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply cmd/sovereignnode/incomingHeader/incomingHeaderProcessor.go:24]: Right, renamed it to MainChainNotarizationStartRound

File: cmd/sovereignnode/incomingHeader/incomingHeaderProcessor.go [hunk 6]
```diff
@@ -20,11 +21,13 @@ type ArgsIncomingHeaderProcessor struct {
...
 }
 
 type incomingHeaderProcessor struct {
 	scrProc            *scrProcessor
 	extendedHeaderProc *extendedHeaderProcessor
+	startRound         uint64
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply cmd/sovereignnode/incomingHeader/incomingHeaderProcessor.go:30]: Right, renamed it to MainChainNotarizationStartRound

File: cmd/sovereignnode/sovereignNodeRunner.go (hunk 1 above)
Comment (mariusmihaic) [reply cmd/sovereignnode/sovereignNodeRunner.go:1676]: Right, renamed it to MainChainNotarizationStartRound

File: config/sovereignConfig.go (hunk 2 above)
Comment (mariusmihaic) [reply config/sovereignConfig.go:10]: Right, added it

File: cmd/sovereignnode/incomingHeader/incomingHeaderProcessor.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply cmd/sovereignnode/incomingHeader/incomingHeaderProcessor.go:30]: Right, renamed it to MainChainNotarizationStartRound

File: config/sovereignConfig.go (hunk 3 above)
Comment (mariusmihaic) [reply config/sovereignConfig.go:9]: Right, added it

**Review by sasurobert — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**